npm_and_yarn/spec/dependabot/npm_and_yarn/metadata_finder_spec.rb

```diff
@@ -192,6 +192,89 @@
       specify { expect { finder.source_url }.to raise_error(JSON::ParserError) }
     end
 
+    context "for a package with private registry source" do
+
+      let(:dependency) do
+        Dependabot::Dependency.new(
+          name: dependency_name,
+          version: "1.0",
+          requirements: [{
+            file: "package.json",
+            requirement: "^1.0",
+            groups: [],
+            source: {
+              type: "private_registry",
+              url: "https://npm.private-registry.io/dependabot"
+            }
+          }],
+          package_manager: "npm_and_yarn"
+        )
+      end
+
+      let(:npm_all_versions_response) { fixture("npm_responses", "etag.json")}
+      let(:npm_latest_version_response) {fixture("npm_responses", "etag-1.0.0.json")}
+
+      context 'using Basic authorization' do
+        before do
+          stub_request(:get, "https://npm.private-registry.io/dependabot/etag/latest").
+            with(headers: { "Authorization" => "Basic #{encoded_token}" }).
+            to_return(status: 200, body: npm_latest_version_response)
+        end
+
+        let(:auth_token) {":secretToken"}
```

Comment:
Move these let statements above.
```suggestion
        let(:auth_token) {":secretToken"}
```